### PR TITLE
Eric Garlock - UI Design Patterns Challenge

### PR DESCRIPTION
**Pull request by testgarlock**
Thank you!

### CODE REVIEW

File: MVVM Module/Source/Feed Presentation/FeedViewModel.swift [hunk 1]
```diff
@@ -20,12 +20,16 @@ final class FeedViewModel {
 	
 	var onLoadingStateChange: Observer<Bool>?
 	var onFeedLoad: Observer<[FeedImage]>?
+	var onFeedError: Observer<String>?
```
Comment (caiozullo) [inline MVVM Module/Source/Feed Presentation/FeedViewModel.swift:23]: `onFeedError: Observer<String> ` could be `onErrorStateChange: Observer<String?>`, so the ViewModel decides which message to show and when to show/hide errors (instead of the view controller). It's often easier to manage state when all changes are centralized.

For example, in the `loadFeed()` method, the ViewModel can call  `onErrorStateChange(.none)` to hide the error before loading the feed. And when there's an error: `onErrorStateChange(Localized.Feed.errorMessage)`.

File: MVVM Module/Source/Feed UI/Controllers/FeedViewController.swift [hunk 2]
```diff
@@ -27,11 +30,15 @@ public final class FeedViewController: UITableViewController, UITableViewDataSou
 		title = viewModel?.title
 		viewModel?.onLoadingStateChange = { [weak self] isLoading in
 			if isLoading {
+				self?.errorView?.hideMessage()
```
Comment (caiozullo) [inline MVVM Module/Source/Feed UI/Controllers/FeedViewController.swift:33]: The ViewModel should decide when to hide/show errors. See my other comment.

**Review by caiozullo — COMMENTED** [review]
Hi @testgarlock,

Congratulations, you've passed the challenge 🥇

- All tests are passing ✅
- Your code is clean and handles edge cases ✅
- The UI and animations look and behave as expected ✅
- Light and Dark mode support ✅
- Added an extra language ✅

I left some review comments. Feel free to push changes.

Great job!
